src/s2wasm-main.cpp

```diff
@@ -108,7 +108,7 @@ int main(int argc, const char *argv[]) {
           : 0;
   if (options.debug) std::cerr << "Global base " << globalBase << '\n';
 
-  Linker linker(globalBase, stackAllocation, initialMem, maxMem,
+  Linker linker((Address)globalBase, (Address)stackAllocation, (Address)initialMem, (Address)maxMem,
```

Comment:
Yeah, a utility function seems fine. For eventual wasm64 there would be a couple options. We could template things on the address size, or just make all the address fields 64 bit and dynamically check (or even do something different for e.g. the interpreter vs the linker if we wanted).